src/Container/ApplicationFactory.php

```diff
@@ -165,9 +166,20 @@ private function injectRoutes(Application $app, ContainerInterface $container)
             $methods = (isset($spec['allowed_methods']) && is_array($spec['allowed_methods']))
                 ? $spec['allowed_methods']
                 : null;
-            $name    = isset($spec['name']) ? $spec['name'] : null;
+            $name = isset($spec['name']) ? $spec['name'] : null;
             $methods = (null === $methods) ? Route::HTTP_METHOD_ANY : $methods;
-            $route   = new Route($spec['path'], $spec['middleware'], $methods, $name);
+
+            if (!is_callable($spec['middleware']) && is_array($spec['middleware'])) {
+                $middlewarePipe = new MiddlewarePipe();
+
+                foreach ($spec['middleware'] as $middleware) {
+                    $middlewarePipe->pipe(is_callable($middleware) ? $middleware : $container->get($middleware));
+                }
+
+                $spec['middleware'] = $middlewarePipe;
+            }
```

Comment:
Interestingly, this comment doesn't show on the website; I think because
its on code you changed in a later commit. Hopefully you'll see this
message!

Yes, I think we should add this into the current pull request, as it makes
the behavior and configuration syntax consistent.

On Tue, Nov 24, 2015 at 11:28 PM, Michael Moussa <notifications@github.com>
wrote:

> In src/Container/ApplicationFactory.php
> <https://github.com/zendframework/zend-expressive/pull/192#discussion_r45830835>
> :
>
> >              $methods = (null === $methods) ? Route::HTTP_METHOD_ANY : $methods;
> > -            $route   = new Route($spec['path'], $spec['middleware'], $methods, $name);
> > +
> > +            if (!is_callable($spec['middleware']) && is_array($spec['middleware'])) {
> > +                $middlewarePipe = new MiddlewarePipe();
> > +
> > +                foreach ($spec['middleware'] as $middleware) {
> > +                    $middlewarePipe->pipe(is_callable($middleware) ? $middleware : $container->get($middleware));
> > +                }
> > +
> > +                $spec['middleware'] = $middlewarePipe;
> > +            }
>
> @weierophinney <https://github.com/weierophinney> I noticed that the
> changes thus far would not (yet) allow this type of array configuration in
> the pre- and post-routing pipelines. Of course, one could just specify them
> separately, like this:
>
> return [    'middleware_pipeline' => [        'pre_routing' => [            [                'middleware' => 'Foo',                'path' => '/foobar',            ],            [                'middleware' => 'Bar',                'path' => '/foobar',            ],            [                'middleware' => 'Foobar',                'path' => '/foobar',            ],        ],    ],];
>
> but something like this would be much more concise and potentially easier
> to read:
>
> return [    'middleware_pipeline' => [        'pre_routing' => [            [                'middleware' => [                    'Foo',                    'Bar',                    'FooBar'                ],                'path' => '/foobar',            ],        ],    ],];
>
> Do you agree? Would you want to try to cram that into this PR as well or
> do it as a future PR hopefully still targeting 1.0.0?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-expressive/pull/192/files#r45830835>
> .
>



-- 
Matthew Weier O'Phinney
matthew@weierophinney.net
https://mwop.net/